pymepix/processing/logic/packet_processor.py

```diff
@@ -0,0 +1,302 @@
+from ctypes import c_bool
+
+import numpy as np
+from pymepix.core.log import Logger
+from pymepix.processing.logic.processing_parameter import ProcessingParameter
+
+from pymepix.processing.logic.processing_step import ProcessingStep
+
+
+class PixelOrientation(IntEnum):
+    """Defines how row and col are intepreted in the output"""
+
+    Up = 0
+    """Up is the default, x=column,y=row"""
+    Left = 1
+    """x=row, y=-column"""
+    Down = 2
+    """x=-column, y = -row """
+    Right = 3
+    """x=-row, y=column"""
+
+
+class PacketProcessor(ProcessingStep):
+    def __init__(self, handle_events=True, event_window=(0.0, 10000.0), position_offset=(0, 0), 
+                orientation=PixelOrientation.Up, start_time=0, timewalk_lut=None, parameter_wrapper_class=ProcessingParameter):
+    
+        super().__init__("PacketProcessor")
+        self._handle_events = parameter_wrapper_class(handle_events)
+        event_window_min, event_window_max = event_window
+        self._event_window_min = parameter_wrapper_class(event_window_min)
+        self._event_window_max = parameter_wrapper_class(event_window_max)
+        self._orientation = orientation
+        self._x_offset, self._y_offset = position_offset
+        self._start_time =  start_time
+        self._timewalk_lut = timewalk_lut
+
+        self._trigger_counter = 0
+
+        self.clearBuffers()
+
+    @property
+    def event_window(self):
+        return (self._event_window_min.value, self._event_window_max.value)
+
+    @event_window.setter
+    def event_window(self, event_window):
+        event_window_min, event_window_max = event_window
+        self._event_window_min.value = event_window_min
+        self._event_window_max.value = event_window_max
+
+    @property
+    def handle_events(self):
+        return self._handle_events.value
+
+    @handle_events.setter
+    def handle_events(self, handle_events):
+        self._handle_events.value = handle_events
+
+    def process(self, data):
+        packet_view = memoryview(data)
+        packet = np.frombuffer(packet_view[:-8], dtype=np.uint64)
+        # needs to be an integer or "(ltime >> 28) & 0x3" fails
+        longtime = int(np.frombuffer(packet_view[-8:], dtype=np.uint64)[0])
+
+        event_data, pixel_data, timestamps = None, None, None
+        if len(packet) > 0:
+
+            header = ((packet & 0xF000000000000000) >> 60) & 0xF
+            subheader = ((packet & 0x0F00000000000000) >> 56) & 0xF
+
+            pixels = packet[np.logical_or(header == 0xA, header == 0xB)]
+            triggers = packet[
+                np.logical_and(
+                    np.logical_or(header == 0x4, header == 0x6), subheader == 0xF
+                )
+            ]
+
+            if pixels.size > 0:
+                pixel_data = self.process_pixels(np.int64(pixels), longtime)
+
+                if triggers.size > 0:
+                    self.process_triggers(np.int64(triggers), longtime)
+
+                if self.handle_events:
```

Comment:
I can not really answer this as I did just copy it over from the old code. My assumption was that handle events is related to tof and centroid calculation and all the other things are also required for toa processing. Therefore I assume we can not simply check this at the beginning of the method as this would break toa processing.